### ISSUE TITLE
Allow channels to link to library resources by name as well as ID

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#3713]
Say a channel is using a specific directory resource on one server. You have that resource set up on two servers with the same name, but behind the scenes they have different IDs. When you export the channel and import it on the other server, the resource is no longer selected, because it's linked by ID instead of name. It would be helpful if that link in the channel model included both the ID and name, so that on import it could try the ID and fall back to the name if needed.

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3830
Reporter: narupley
Created: 2015-11-05T09:13:14.000-0800

**Comment by rbeckman-nextgen** [comment]
Revision 7914: Resource names are now stored alongside IDs in the models, so that on import into a different server, the name can be used as a fallback to link to resources. Whenever a channel/connector is saved/imported/exported, the Administrator will automatically update any resource names in the map.

Imported Comment. Original Details:
Author: narupley
Created: 2016-02-16T12:16:02.000-0800

**Comment by rbeckman-nextgen** [comment]
Verified that resource dependencies/links are restored using the name as a backup.

I added 5 resources to an instance of MC 3.4 and created a separate 3.4 instance with the same exact 5 resources. I was able to export the channel from one instance to the other and verified that all links were restored. This obviously works for exporting the same channel and modifying the resource IDS directly in the XML. I also tried setting links on each individual level (Channel, Scripts, Source Connector, Destination Connector) and verified that these were also restored correctly.

Imported Comment. Original Details:
Author: jaysenp
Created: 2016-03-30T18:29:53.000-0700
